mtls_server/auth.py

```diff
@@ -0,0 +1,93 @@
+import base64
+import os
+import time
+import pprint
+import json
+from functools import wraps
+
+from flask import request
+from flask import g
+from flask import current_app
+
+from .logger import logger
+from .utils import error_response
+from .utils import time_in_range
+from .utils import write_sig_to_file
+
+
+pp = pprint.PrettyPrinter(indent=4)
+
+
+class SignatureTimestampOutOfBoundsException(Exception):
+    pass
+
+
+class MissingTokenException(Exception):
+    pass
+
+
+class BadSignatureException(Exception):
+    pass
+
+
+def login_required(f):
+    @wraps(f)
+    def login_required_wrap(*args, **kwargs):
+        allowed_tokens = ['PGP-SIG']
+        g.is_admin = False
+
+        if not request.headers['Authorization']:
+            return error_response("authentication required", 401)
+
+        token_type, token = request.headers['Authorization'].split(' ')
+
+        if token_type not in allowed_tokens:
+            return error_response("authentication required", 401)
+
+        if token_type == "PGP-SIG":
+            data = request.get_data()
```

Comment:
From docs:
> Usually it’s a bad idea to call this method without checking the content length first as a client could send dozens of megabytes or more to cause memory problems on the server.

What are you doing to protect against that?
I also notice there is no timeout: would this be vulnerable to a slow loris attack?

Reply:
There's a config we can pass through, I saw that in the docs and I there was some default but I may have also looked at the nginx docs to see if there is a default there (which there is, it's 1Mb). But then we'd be making the assumption that someone is using nginx or another proxy in front of this. 

Good catch

Reply:
I actually did think of this: https://github.com/drGrove/mtls-server/pull/56/files#diff-8a171bbdfde7635fb9f28f0fd9840c48b1800d471ab36b64fe9714ee72a37249R41

I just didn't make it a default configuration in the kustomize